### PR TITLE
fix(esp-sync): sync Connected Account field

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Restores the syncing of the `NP_Connected Account` field, which was dropped at some point in the past.

### How to test the changes in this Pull Request:

1. In WP admin > Engagement > Reader Activation > Advanced Settings, enable the **Connected Account** field for syncing.
2. As a reader, complete a purchase via Woo checkout (modal or not) using a real live Gmail address.
3. Log out, then log in via Google SSO using the Gmail address.
4. In the connected ESP, confirm that contact's metadata has the following:
  - `NP_Registration Method` is `woocommerce`
  - `NP_Connected Account` is `google`
5. In a new session, register a new account via Google SSO using a different Gmail address.
6. Complete another purchase via Woo checkout.
7. In the connected ESP, confirm that contact's metadata has the following:
  - `NP_Registration Method` is `google`
  - `NP_Connected Account` is `google`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208262205728462

### CODE REVIEW

File: includes/reader-activation/sync/class-metadata.php [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 namespace Newspack\Reader_Activation\Sync;
 
+use Error;
```
Comment (miguelpeixe) [inline includes/reader-activation/sync/class-metadata.php]: What's this for?

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
It worked well with `google` as the registration method.

When registering via the modal checkout, the user is created without the `np_reader_registration_method` meta, maybe an existing bug on trunk?

File: includes/reader-activation/sync/class-metadata.php [hunk 2]
```diff
@@ -330,10 +332,25 @@ public static function normalize_contact_data( $contact ) {
...
+			$connected_account               = ! empty( $metadata['connected_account'] ) ? $metadata['connected_account'] : \get_user_meta( $user->ID, Reader_Activation::CONNECTED_ACCOUNT, true );
+			if ( ! empty( $registration_method ) ) {
+				$metadata['registration_method'] = $registration_method;
+			}
+			if ( $connected_account && in_array( $connected_account, Reader_Activation::SSO_REGISTRATION_METHODS ) ) {
+				$metadata['connected_account'] = $connected_account;
+			} elseif ( $registration_method && in_array( $registration_method, Reader_Activation::SSO_REGISTRATION_METHODS ) ) {
+				$metadata['connected_account'] = $registration_method;
+			}
+		}
```
Comment (miguelpeixe) [inline includes/reader-activation/sync/class-metadata.php]: Given this method has so many different steps, I think we'd have better readability by running this in a separate method.

Comment (miguelpeixe) [reply includes/reader-activation/sync/class-metadata.php]: We should probably do the same with the UTM capturing bit.

File: includes/reader-activation/sync/class-metadata.php (hunk 1 above)
Comment (dkoo) [reply includes/reader-activation/sync/class-metadata.php]: That's a mistake—my VSCode likes to "helpfully" try to insert dependencies if it thinks I'm missing any (such as if I'm typing `error_log`)

File: includes/reader-activation/sync/class-metadata.php (hunk 2 above)
Comment (dkoo) [reply includes/reader-activation/sync/class-metadata.php]: Good idea! 7037b6c4 creates separate methods for both, and also refactors the `registration_method` and `connected_account` checks to use the class's built-in helpers for `has_key()` and `get_key_value()`.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
Thank you for the revisions! Connected Account and UTM tested well.

The issue I had with the `woocommerce` registration method is no longer reproducible, not sure what happened there.

I did find another issue, with Google oauth. It seems the referrer is not passed along so it's unable to set the registration page and, naturally, the UTM params as well. Do you think it's something we can address in this PR? If not I can approve and we address it separately.

**Comment by dkoo** [comment]
> I did find another issue, with Google oauth. It seems the referrer is not passed along so it's unable to set the registration page and, naturally, the UTM params as well. Do you think it's something we can address in this PR? If not I can approve and we address it separately.

25294c7 should fix the Registration Page and Signup UTM fields for SSO registrations!

**Review by miguelpeixe — APPROVED** [review]
Thanks for revising! Working well with Google + UTM!

**Pull request merged**